### PR TITLE
fix(es/resolver): treat switch stmt as block scope

### LINKED ISSUE #3266: False error inside `switch`: Cannot reassign to a variable declared with `const`
### Describe the bug

Running the input code works fine in node.js & chrome console:
CleanShot 2022-01-14 at 12 48 10@2x

But throwing it into https://swc.rs/playground yields:

```log
error: Cannot reassign to a variable declared with `const`
 --> input.js:1:14
  |
1 | function bar(fooArg) {
  |              ^^^^^^
  |
note: fooArg was declared here
 --> input.js:3:22
  |
3 |     case true: const fooArg = 'test'; return 'bar() result';
  |                      ^^^^^^
```

### Input code

```typescript
function bar(fooArg) {
  switch (true) {
    case true: const fooArg = 'test'; return 'bar() result';
  }
}
console.log(bar());
```

### Config

_No response_

### Playground link

https://play.swc.rs/?version=1.2.129&code=H4sIAAAAAAAAAyXMQQqAIBCF4b2neDv1CkWLjmKiJcQIMyMtwruntHwf%2FC83iloq4Qjscq07nx6vAeQpGi845ZZ%2BAWKQhAkLYiVR%2FAE2WE2idgUnbUyw882PJe0ePOJu%2BgeX5CD4awAAAA%3D%3D&config=H4sIAAAAAAAAA0WMSwrDMAxE76K1F02gXeQOPYRwleDiH5ICNcZ3r11cshtm3rwKb7GwVcjIQjySlKj4gQ3IBhTLLiuYjvVqRy%2FUDCjyQToQWW%2FLvc8%2BJaEJGAguur0MmU0hM4lcE8bD%2F8nWXSG9zlFU0JLp53xAuxzz5%2BQ5QeWT2he35aBbuAAAAA%3D%3D

### Expected behavior

Should not throw an error on given snippet. 

The `switch {}` creates a new scope, just like `if` does. And swc already [handles the `if` situation as node.js does](https://play.swc.rs/?version=1.2.129&code=H4sIAAAAAAAAAyWLQQrAIAwE775ib%2BobSg99SiuxCCWBmJyKf6%2FF4wwz1blYE8Z1aqoih94ZbwBaRTJ1WgQU4W5YBXZEo25xg5K5MuK%2F50ndn6nnMcL4AH1TqK9cAAAA&config=H4sIAAAAAAAAA0WMSwrDMAxE76K1F02gXeQOPYRwleDiH5ICNcZ3r11cshtm3rwKb7GwVcjIQjySlKj4gQ3IBhTLLiuYjvVqRy%2FUDCjyQToQWW%2FLvc8%2BJaEJGAguur0MmU0hM4lcE8bD%2F8nWXSG9zlFU0JLp53xAuxzz5%2BQ5QeWT2he35aBbuAAAAA%3D%3D)

### Actual behavior

_No response_

### Version

1.2.129

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
- fix #3266

### CODE REVIEW

File: crates/swc_ecma_transforms_base/src/resolver/mod.rs [hunk 1]
```diff
@@ -488,6 +488,19 @@ impl<'a> VisitMut for Resolver<'a> {
...
+        s.discriminant.visit_mut_with(self);
+
+        let child_mark = Mark::fresh(Mark::root());
+
+        let mut child_folder = Resolver::new(
+            Scope::new(ScopeKind::Block, child_mark, Some(&self.current)),
+            self.handle_types,
+        );
+
+        s.visit_mut_children_with(&mut child_folder);
```
Comment (kdy1) [inline crates/swc_ecma_transforms_base/src/resolver/mod.rs:501]: I think this should be `s.cases.visit_mut_with(&mut child_folder);`

Comment (magic-akari) [reply crates/swc_ecma_transforms_base/src/resolver/mod.rs:501]: oh, thanks

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Comment by kdy1** [comment]
Can you run `UPDATE=1 cargo test` from the minifier crate? Seems like CI failed due to it.

**Review by kdy1 — APPROVED** [review]

**Pull request merged**